### PR TITLE
Feature/5

### PR DESCRIPTION
**Pull request by caleeli**

### CODE REVIEW

File: src/ProcessMaker/Nayra/Bpmn/CloseExceptionTransition.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+use ProcessMaker\Nayra\Contracts\Bpmn\TransitionInterface;
+
+/**
+ * Transition rule for an activity in FAILING state.
+ *
+ * @package ProcessMaker\Nayra\Bpmn
+ */
+class CloseExceptionTransition implements TransitionInterface
+{
+
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/CloseExceptionTransition.php]: Remove this extra line

File: src/ProcessMaker/Nayra/Bpmn/EndEventTrait.php [hunk 2]
```diff
@@ -44,13 +42,13 @@
     public function buildTransitions(RepositoryFactoryInterface $factory)
     {
         $this->setFactory($factory);
-        $this->endState = new State($this);
+        $this->endState = new State($this, 'CLOSING');
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/EndEventTrait.php]: Try to put the string 'CLOSING' in a constant

File: src/ProcessMaker/Nayra/Bpmn/ExceptionTransition.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+use ProcessMaker\Nayra\Contracts\Bpmn\TransitionInterface;
+
+/**
+ * Transition rule when a exception is catch.
+ *
+ * @package ProcessMaker\Nayra\Bpmn
+ */
+class ExceptionTransition implements TransitionInterface
+{
+
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/ExceptionTransition.php]: Remove this extra line

File: src/ProcessMaker/Nayra/Bpmn/ExceptionTransition.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Condition required to transit the element.
+     *
+     * @param TokenInterface $token
+     *
+     * @return bool
+     */
+    public function assertCondition(TokenInterface $token)
+    {
+        return $token->getProperty('STATUS') === ActivityInterface::TOKEN_STATE_FAILING;
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/ExceptionTransition.php]: Try to put the string 'STATUS' in a constant

File: src/ProcessMaker/Nayra/Bpmn/InclusiveGatewayTrait.php [hunk 5]
```diff
@@ -50,10 +44,10 @@ public function buildTransitions(RepositoryFactoryInterface $factory)
      * @return StateInterface
      */
     public function getInputPlace() {
-        $incomingPlace=new State($this);
+        $incomingPlace=new State($this, 'INCOMMING');
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/InclusiveGatewayTrait.php]: Try to put the 'INCOMMING' string in a constant

File: src/ProcessMaker/Nayra/Bpmn/InclusiveGatewayTrait.php [hunk 6]
```diff
@@ -82,7 +76,7 @@ protected function buildConnectionTo(FlowNodeInterface $target)
      * @return $this
      */
     protected function buildConditionedConnectionTo(FlowNodeInterface $target, callable $condition, $default=false) {
-        $outgoingPlace = new State($this);
+        $outgoingPlace = new State($this, 'OUTGOING');
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/InclusiveGatewayTrait.php]: Try to put the 'OUTGOING' string in a constant

File: src/ProcessMaker/Nayra/Bpmn/StateTrait.php [hunk 7]
```diff
@@ -31,10 +29,18 @@
...
 
-    protected function initPlaceBehavior(EntityInterface $owner)
+    /**
+     * State name.
+     *
+     * @var string
+     */
+    private $name;
+
+    protected function initPlaceBehavior(EntityInterface $owner, $name)
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/StateTrait.php]: Add the documentation of the method

File: src/ProcessMaker/Nayra/Bpmn/Transition.php [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+use ProcessMaker\Nayra\Contracts\Bpmn\TransitionInterface;
+
+/**
+ * Transition rule that always pass the token.
+ *
+ * @package ProcessMaker\Nayra\Bpmn
+ */
+class Transition implements TransitionInterface
+{
+
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/Transition.php]: Remove this extra line

File: tests/ProcessMaker/Models/ActivityWithException.php [hunk 9]
```diff
@@ -18,13 +18,14 @@ class ActivityWithException implements ActivityInterface
...
+     * Configure the activity to go to a FAILING status when activated.
      *
-     * @throws ActivityWorkException
      */
-    public function work()
+    public function initActivity()
     {
-        throw new ActivityWorkException();
+        $this->attachEvent(ActivityInterface::EVENT_ACTIVITY_ACTIVATED, function ($self, TokenInterface $token) {
+            $token->setProperty('STATUS', ActivityInterface::TOKEN_STATE_FAILING);
```
Comment (danloa) [inline tests/ProcessMaker/Models/ActivityWithException.php]: Try to put the 'STATUS' string in a constant

File: src/ProcessMaker/Nayra/Bpmn/Collection.php [hunk 10]
```diff
@@ -201,4 +201,14 @@ public function seek($index)
     {
         $this->index = $index;
     }
+
+    /**
+     * Converts the collection to an array
+     *
+     * @return void
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/Collection.php]: the return type of the documentation is incorrect, it is an array

File: src/ProcessMaker/Nayra/Bpmn/ExceptionTransition.php (hunk 4 above)
Comment (caleeli) [reply src/ProcessMaker/Nayra/Bpmn/ExceptionTransition.php]: This will be changed to a setter in feature/7

**Pull request merged**